biomaj/bank.py

```diff
@@ -525,10 +527,36 @@ def save_session(self):
 
         self.bank = self.banks.find_one({'name': self.name})
 
+    def format_release(self, release, reverse=False):
+        """
+        Check the release does not contains any dot '.' character that should make MongoDB complain
+
+        :param release: release number to check
+        :type release: str
+        :param reverse: Do the reverse replacement
+        :type reverse: bool
+        :return: Release string with dot '.' replaced with character set in config (release.replace)
+        :rtype: str
+        """
+        if not self.config.get('release.replacedot'):
```

Comment:
yeap, only impact in workflow.py should be in release = reg.group(X) in
following code


            if cf.get('release.regexp') is None or not
cf.get('release.regexp'):
                # Try to get from regexp in file name
                rel = re.search(cf.get('release.file'),
release_downloader.files_to_download[0]['name'])
                if rel is None:
                    logging.error('release.file defined but does not match
any file')
                    return False
                release = rel.group(1)
            else:
                # Download and extract
                tmp_dir = tempfile.mkdtemp('biomaj')
                release_downloader.mkdir_lock = DownloadThread.MKDIR_LOCK
                rel_files = release_downloader.download(tmp_dir)
                rel_file = open(tmp_dir + '/' + rel_files[0]['name'])
                rel_content = rel_file.read()
                rel_file.close()
                shutil.rmtree(tmp_dir)
                rel = re.search(cf.get('release.regexp'), rel_content)
                if rel is None:
                    logging.error('release.regexp defined but does not
match any file content')
                    return False
                # If regexp contains matching group, else take whole match
                if len(rel.groups()) > 0:
                    release = rel.group(1)
                else:
                    release = rel.group(0)

Le mar. 26 avr. 2016 à 17:48, olivier sallou <olivier.sallou@gmail.com> a
écrit :

> therer should be a single place (or like) for this in workflow.py, I gonna
> have a look at the code
>
> Le mar. 26 avr. 2016 à 17:46, Emmanuel Quevillon <notifications@github.com>
> a écrit :
>
>> In biomaj/bank.py
>> <https://github.com/genouest/biomaj/pull/44#discussion_r61110671>:
>>
>> > @@ -525,10 +527,36 @@ def save_session(self):
>> >
>> >          self.bank = self.banks.find_one({'name': self.name})
>> >
>> > +    def format_release(self, release, reverse=False):
>> > +        """
>> > +        Check the release does not contains any dot '.' character that should make MongoDB complain
>> > +
>> > +        :param release: release number to check
>> > +        :type release: str
>> > +        :param reverse: Do the reverse replacement
>> > +        :type reverse: bool
>> > +        :return: Release string with dot '.' replaced with character set in config (release.replace)
>> > +        :rtype: str
>> > +        """
>> > +        if not self.config.get('release.replacedot'):
>>
>> yes I always forget about default value available as argument.
>> I've check in workflow.py to directly set/replace the release using
>> format_release.
>> It also looks that setting of remote release is done in multiple
>> different places:
>>
>> config.py:279:        self.set('remoterelease', '')
>> workflow.py:68:        self.session.config.set('remoterelease', '')
>> workflow.py:401:                    self.session.set('remoterelease', dep.session.get('remoterelease'))
>> workflow.py:517:        self.session.set('remoterelease', release)
>> workflow.py:718:            self.session.config.set('remoterelease', self.session.get('remoterelease'))
>> workflow.py:871:            self.session.set('remoterelease', release)
>> workflow.py:891:        self.session.config.set('remoterelease', self.session.get('remoterelease'))
>> workflow.py:1253:                    session.set('remoterelease', prod['remoterelease'])
>>
>> I'll try to move this modification into workflow.py
>> The thing is we will need a method accessible from other modules and also
>> able to access the config object.
>>
>> —
>> You are receiving this because you commented.
>>
>>
>> Reply to this email directly or view it on GitHub
>> <https://github.com/genouest/biomaj/pull/44/files/d235f544d677e718229ab04b92229485fa50ca28#r61110671>
>>
>